kythe/go/platform/tools/kzip/createcmd/createcmd.go

```diff
@@ -39,117 +36,147 @@ import (
 type createCommand struct {
 	cmdutil.Info
 
-	uri     string
-	output  string
-	source  string
-	details string
+	output string
+	rules  vnameRules
+
+	uri          kytheURI
+	source       repeatedString
+	inputs       repeatedString
+	hasError     bool
+	argument     repeatedString
+	outputKey    string
+	workingDir   string
+	entryContext string
+	environment  repeatedEnv
+	details      repeatedAny
 }
 
 // New creates a new subcommand for merging kzip files.
 func New() subcommands.Command {
 	return &createCommand{
-		Info: cmdutil.NewInfo("create", "create simple kzip archives", `-uri <uri> -output <path> -source <path> required_input*
+		Info: cmdutil.NewInfo("create", "create simple kzip archives", `[options] -- arguments*
 
-Construct a kzip file consisting of a single input file name by -source.
-The resulting file is written to -output, and the vname of the compilation
-will be attributed to the values specified within -uri.
+Construct a kzip file written to -output with the vname specified by -uri.
+Each of -source_file, -required_input and -details may be specified multiple times with each
+occurrence of the flag being appended to the corresponding field in the compilation unit.
 
-Additional required inputs, if any, can be provided as positional parameters.
+Any additional positional arguments are included as arguments in the compilation unit.
 `),
 	}
 }
 
 // SetFlags implements the subcommands interface and provides command-specific flags
 // for creating kzip files.
 func (c *createCommand) SetFlags(fs *flag.FlagSet) {
-	fs.StringVar(&c.uri, "uri", "", "A Kythe URI naming the compilation unit")
-	fs.StringVar(&c.output, "output", "", "Path for output kzip file")
-	fs.StringVar(&c.source, "source", "", "Path for input source file")
-	fs.StringVar(&c.details, "details", "", "JSON-encoded Any messages to embed as compilation details.")
+	fs.StringVar(&c.output, "output", "", "Path for output kzip file (required)")
+	fs.Var(&c.rules, "rules", "Path to vnames.json file (optional)")
+
+	fs.Var(&c.uri, "uri", "A Kythe URI naming the compilation unit VName (required)")
+	fs.Var(&c.source, "source_file", "Repeated paths for input source files (required)")
+	fs.Var(&c.inputs, "required_input", "Repeated paths for additional required inputs (optional)")
```

Comment:
Add a note that these should not intersect with `--source_file` or there will be duplicate required inputs in the compilation.

Reply:
Ideally the tool should deduplicate them, right?

Reply:
indeed; changed this to stringset flags